### PR TITLE
CI is enforced over java code to demand basic code quality

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Part of #107

### CODE REVIEW

File: extractor/build.gradle [hunk 1]
```diff
@@ -41,4 +42,49 @@ test {
         showStandardStreams = true
         events "passed", "skipped", "failed"
     }
-}
\ No newline at end of file
+}
+
+// PMD Configuration
+pmd {
+    toolVersion = "6.55.0"
```
Comment (romani) [inline extractor/build.gradle:49]: Use latest, 7.x

Comment (romani) [reply extractor/build.gradle:49]: https://github.com/checkstyle/checkstyle/blob/6f87a86f39b35e8c09d528e664b775943345560d/pom.xml#L213

File: extractor/build.gradle [hunk 2]
```diff
@@ -1,6 +1,8 @@
 plugins {
     id 'application'
     id 'java'
+    id 'pmd'
+    id 'com.github.spotbugs' version '5.0.13'
```
Comment (romani) [inline extractor/build.gradle:5]: https://plugins.gradle.org/plugin/com.github.spotbugs

Please use latest.

File: extractor/config/spotbugs/spotbugs-exclude.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<FindBugsFilter>
+    <Match>
+        <!-- to investigate: not clear why catch all is happening -->
```
Comment (romani) [inline extractor/config/spotbugs/spotbugs-exclude.xml:4]: First version should have no excludes

File: extractor/config/pmd/pmd.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0
+                             http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+    <description>
+        Custom PMD ruleset for the extractor project
+    </description>
+
+    <rule ref="category/java/bestpractices.xml">
+        <exclude name="GuardLogStatement"/>
+        <exclude name="UnusedPrivateMethod"/>
```
Comment (romani) [inline extractor/config/pmd/pmd.xml:13]: Let's use all , and exclude only if it becomes problematic for us.

**Review by romani — CHANGES_REQUESTED** [review]
Items 

I highly recommend to do each tool in separate PR.

**Comment by romani** [comment]
@relentless-pursuit , please reply done to each review comments, I do understand if you ready for review or not

**Comment by romani** [comment]
As you have no changes in code it means that all such tools are inactive and executed during build.

File: extractor/config/pmd/pmd.xml (hunk 4 above)
Comment (relentless-pursuit) [reply extractor/config/pmd/pmd.xml]: Done

File: extractor/config/spotbugs/spotbugs-exclude.xml (hunk 3 above)
Comment (relentless-pursuit) [reply extractor/config/spotbugs/spotbugs-exclude.xml]: Done

File: extractor/build.gradle (hunk 2 above)
Comment (relentless-pursuit) [reply extractor/build.gradle]: Done

File: extractor/build.gradle (hunk 1 above)
Comment (relentless-pursuit) [reply extractor/build.gradle]: Done

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit , please reply done to each review comments, I do understand if you ready for review or not

Sorry, i missed. I was focused on other issue.

**Comment by relentless-pursuit** [comment]
> As you have no changes in code it means that all such tools are inactive and executed during build.

Yes, the build failed first. And, I have disabled it for now. I hope you are fine with this. I guess, these changes can be done later. I am just focussing on the high priority ones right now like conv to java project, which seems to me a huge change.

**Comment by relentless-pursuit** [comment]
> Items
> 
> I highly recommend to do each tool in separate PR.

About the tools to be done in separate, I was hoping can we create a framework for this. I guess the code refactoring can be taken up simultaneously when the major issues are done.

**Comment by romani** [comment]
>ignoreFailures = false

We are not ignoring. I think your configs are simply ignored at all, not working at all.

**Comment by romani** [comment]
ok, I got it.

CI is green because we just do `test`
```
 cd extractor
  ./gradlew clean build test run --args="../.ci-temp/checkstyle" --info --stacktrace --console=plain \
    -Dorg.gradle.logging.level=DEBUG \
    -Dtest.single=MainsLauncherTest
```

**Comment by romani** [comment]
@relentless-pursuit , please send code cleanups, postpone complicated updates or later and share with me complicated parts, there might be some suppressions possible

**Pull request merged**